src/app/navigation/utils.ts

```diff
@@ -109,3 +114,26 @@ export const findPathForParams = (params: object, paths: string[]) => {
       paramsInPath.every(({name}) => paramKeys.includes(name.toString()));
   });
 };
+
+export const getScrollTargetFromQuery = (
```

Comment:
I like this approach and i feel like it should be the same for generating a query from scroll target.